packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2512,29 +2612,73 @@ export function writeClientRenderBoundaryInstruction(
   }
 
   writeChunk(destination, boundaryID);
-  writeChunk(destination, clientRenderScript1A);
+  if (scriptFormat) {
+    // " needs to be inserted for scripts, since ArgInterstitual does not contain
+    // leading or trailing quotes
+    writeChunk(destination, clientRenderScript1A);
+  }
+
   if (errorDigest || errorMessage || errorComponentStack) {
-    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
-    writeChunk(
-      destination,
-      stringToChunk(escapeJSStringsForInstructionScripts(errorDigest || '')),
-    );
+    if (scriptFormat) {
+      // ,"JSONString"
+      writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+      writeChunk(
+        destination,
+        stringToChunk(escapeJSStringsForInstructionScripts(errorDigest || '')),
+      );
+    } else {
+      // " arg1="EscapedString
+      writeChunk(destination, clientRenderData2);
+      writeChunk(
+        destination,
+        stringToChunk(escapeTextForBrowser(errorDigest || '')),
+      );
+    }
```

Comment:
might consider inverting the conditionals here too.

If we are using script format we need positional arguments so we need to check for digest/message/stack even if the digest is empty b/c we need to consume a slot to get the stack for instance to show up where it should

For runtime script we have named arguments and can elide the missing ones since we can infer them to be empty (does require a small change to the runtime so maybe this isn't worth it)

If you check scriptFormat first you can completely change the logic around emitting missing arguments